### PR TITLE
Changes after GitHub review #2

### PR DESCRIPTION
**Pull request by tomchy**
The module path check does not work in CI builds.

### CODE REVIEW

File: zboss/CMakeLists.txt [hunk 1]
```diff
@@ -6,30 +6,35 @@
...
-# Add compile-time definition, indicating which ZBOSS library will be linked
-if (CONFIG_ZIGBEE_ROLE_END_DEVICE)
-  zephyr_compile_definitions(ZB_ED_ROLE)
-endif()
+# Link with interface libraries - zboss and zboss-extenstions
+zephyr_library_link_libraries(zboss-extensions)
+zephyr_library_link_libraries(zboss)
+
+# Link with Zephyr's default target
+zephyr_link_libraries(zboss-extensions)
```
Comment (tejlmand) [inline zboss/CMakeLists.txt]: I don't think this is needed.
What you would need, is to `zephyr_library_link_libraries(zboss-extensions)` in the `nrf/subsys/zigbee/` lib.

If the app needs to link to the zboss-extension, then there should be a KConfig entry with:
```
config APP_LINK_WITH_ZBOSS-EXTENSION
    bool "Link application with ZBoss library"
    default y
    help
        Some help text
```

File: zboss/CMakeLists.txt [hunk 2]
```diff
@@ -38,14 +43,42 @@ if ((NOT DEFINED ZEPHYR_PLATFORM_NCS_MODULE_DIR) OR
...
-    zephyr_link_libraries(${ZBOSS_LIB_PATH}/libzboss.ed.a)
-    toolchain_cc_imacros(${CMAKE_CURRENT_SOURCE_DIR}/include/osif/libzboss_config.ed.h)
+    zephyr_library_import(zboss-libraries ${ZBOSS_LIB_PATH}/libzboss.ed.a)
+    set(LIBZBOSS_CONFIG_FILE "libzboss_config.ed.h")
   else()
     message( FATAL_ERROR "Unsupported Zigbee role. Exiting." )
   endif()
 
+  # Link with Zephyr's default target
+  zephyr_link_libraries(zboss)
```
Comment (tejlmand) [inline zboss/CMakeLists.txt]: Same as above.

**Review by tejlmand — COMMENTED** [review]
Good direction.
You could decide to combine `zboos` and `zboos-extension` into one library.
But it is also nice that they are individual, cause it makes it very clear what is nordic extension.

But to ease the use, you could then consider to do a:
```
target_link_libraries(zboss INTERFACE zboss-extension)
```
(or reverse)
So other libs can simply link to `zboss` and automatically inherit the other.

File: zboss/include/osif/zb_vendor.h [hunk 3]
```diff
@@ -9,6 +9,7 @@
 #ifndef ZB_VENDOR_H__
 #define ZB_VENDOR_H__
 
+#include "libzboss_config_generic.h"
```
Comment (tejlmand) [inline zboss/include/osif/zb_vendor.h]: Wouldn't this suffice ?
```
#ifdef LIBZBOSS_CONFIG_FILE
#include "${LIBZBOSS_CONFIG_FILE}"
#endif
```
and then when using the pre-built lib in the nrfxlib repo, then `LIBZBOSS_CONFIG_FILE` is defined.
But when building the source, `LIBZBOSS_CONFIG_FILE` is not defined, and hence the config header is not used ?

And then you don't need `zboss/include/osif/libzboss_config_generic.h.template ` and the above mentioned lines.

File: zboss/CMakeLists.txt [hunk 4]
```diff
@@ -38,14 +43,42 @@ if ((NOT DEFINED ZEPHYR_PLATFORM_NCS_MODULE_DIR) OR
...
+    include/se
+    include/osif
+  )
+
+  # Define the ZBOSS imported library, that contains only static libraries
   if (CONFIG_ZIGBEE_ROLE_COORDINATOR OR CONFIG_ZIGBEE_ROLE_ROUTER)
-    zephyr_link_libraries(${ZBOSS_LIB_PATH}/libzboss.a)
-    toolchain_cc_imacros(${CMAKE_CURRENT_SOURCE_DIR}/include/osif/libzboss_config.h)
+    zephyr_library_import(zboss-libraries ${ZBOSS_LIB_PATH}/libzboss.a)
+    set(LIBZBOSS_CONFIG_FILE "libzboss_config.h")
```
Comment (tejlmand) [inline zboss/CMakeLists.txt]: did you mean:
```suggestion
target_compile_definitions(zboss INTERFACE LIBZBOSS_CONFIG_FILE=libzboss_config.h)
```

File: zboss/CMakeLists.txt [hunk 5]
```diff
@@ -38,14 +43,42 @@ if ((NOT DEFINED ZEPHYR_PLATFORM_NCS_MODULE_DIR) OR
...
+  # Link with Zephyr's default target
+  zephyr_link_libraries(zboss)
+
+  # Generate libzboss_config_generic, which is included in all files, that uses
+  # ZBOSS API
+  configure_file(
+    ${CMAKE_CURRENT_SOURCE_DIR}/include/osif/libzboss_config_generic.h.template
+    ${CMAKE_CURRENT_SOURCE_DIR}/include/osif/libzboss_config_generic.h
+  )
+endif()
```
Comment (tejlmand) [inline zboss/CMakeLists.txt]: Based on the other comments, I think this could be removed.

**Review by tejlmand — APPROVED** [review]
Good direction: 👍 

Couple of comments regarding the config header.

File: zboss/CMakeLists.txt [hunk 6]
```diff
@@ -38,14 +43,42 @@ if ((NOT DEFINED ZEPHYR_PLATFORM_NCS_MODULE_DIR) OR
...
+
+
+
+# Suppress bitfield compatibility warning
+zephyr_compile_options(-Wno-packed-bitfield-compat)
```
Comment (tomchy) [inline zboss/CMakeLists.txt]: library_compile_options for interface library

File: zboss/CMakeLists.txt (hunk 4 above)
Comment (tomchy) [reply zboss/CMakeLists.txt]: I've included apostrophes too, so you can place this header anywhere you want ;)

File: zboss/include/osif/zb_vendor.h (hunk 3 above)
Comment (tomchy) [reply zboss/include/osif/zb_vendor.h]: Changes applied.

File: zboss/CMakeLists.txt (hunk 2 above)
Comment (tomchy) [reply zboss/CMakeLists.txt]: Fixed. Now application links to zboss, which links to zboss-extensions, as well as zboss-extensions C files, which have autogenerated name.

File: zboss/CMakeLists.txt (hunk 1 above)
Comment (tomchy) [reply zboss/CMakeLists.txt]: I prefer ZBOSS than ZBOSS-EXTENSION to be visible by the user.
Still - this option will be hidden (there is no Zigbee app that does not need ZBOSS lib).

**Comment by tomchy** [comment]
@maciekfabia Can we merge it and continue the review on the official PR?

**Comment by maciekfabia** [comment]
> @maciekfabia Can we merge it and continue the review on the official PR?

merged

**Pull request merged**